### ISSUE TITLE
Optional images GraphQL error

### ISSUE THREAD

**Issue by bobevsky** [spatie/statamic-responsive-images#162]
When you add new entry inside a collection (leave out the image) but there is no images in the assets folder for that collection, GraphQL returns error even though the image is optional.

Steps to reproduce:
1. start with a blank collection
2. remove every image from the assets inside the folder for that collection (if you have any).
3. Try to add a collection entry but don't fill out the images (make them optional if they aren't already)
4. Then try to get the entries from that collection inside GraphQL and this should pop out:

![image](https://user-images.githubusercontent.com/44194428/186097239-c4154567-7aca-4408-a15a-fd12817db1ac.png)

**Comment by ncla** [comment]
I am unable to reproduce your scenario where it throws the exception. No asset selected, asset container is empty, fresh collection with just one entry. Asset is optional.

```graphql
{
    entries(collection: "projects") {
        current_page
        data {
            id
            title
            ... on Entry_Projects_Project {
            	assets_field {
                path
              }
              responsive_field {
                breakpoints {
                  asset {
                    path
                  }
                  srcSet
                }
              }
            }
        }
    }
}
```

returns 

```json
{
  "data": {
    "entries": {
      "current_page": 1,
      "data": [
        {
          "id": "fb403e42-70fb-4707-a79a-589780051bf1",
          "title": "Amazing",
          "assets_field": null,
          "responsive_field": null
        }
      ]
    }
  }
}
```

However if both `assets_field` and `responsive_field` were pointing to an asset that does not exist (changing file location in the entry file), then I do receive the same exception from `responsive_field` field. `assets_field` fails silently and just returns `null`. Theoretically `responsive_field` also could fail silently the same way `assets_field` does upon not found asset.

**Comment by ncla** [comment]
> Still relevant. I found additional issue where if one of the breakpoint assets are not found, then all other ones will fail too, and produce no srcsets at all.

This I could not reproduce anymore and I figured out that #173 resolved it, as before this PR, `src` would be saved as array and contain container ID as well e.g. `test_container::asset.jpg`, which would then weirdly augment the value in GraphQL stuff.

An issue with required validation has been fixed for default breakpoint asset in #164, which could lead control panel users to have such entry data that fails in GraphQL.

Lastly, in todays update v3.1.2, if default breakpoint asset has not been found, then data will return `null` value and an error will be silently logged. Similarly to how Responsive tag will return empty string. And similarly, Asset fieldtype also returns `null` when asset could not be found.
